src/components/survey/reusables.tsx

```diff
@@ -0,0 +1,505 @@
+    const {
+      componentName,
+      reusableMaterialId,
+      usability,
+      unit,
+      description,
+      amount
+    } = newMaterial;
+
+    if (!keycloak?.token || !surveyId) {
+      return;
+    }
+
+    try {
+      const createdReusable = await Api.getSurveyReusablesApi(keycloak.token).createSurveyReusable({
+        surveyId: surveyId,
+        reusable: {
+          componentName: componentName,
+          reusableMaterialId: reusableMaterialId,
+          usability: usability,
+          unit: unit,
+          description: description,
+          amount: amount,
+          metadata: {}
+        }
+      });
+      setSurveyReusables([ ...surveyReusables, createdReusable ]);
+    } catch (error) {
+      errorContext.setError(strings.errorHandling.reusables.create, error);
+    }
+
+    setAddingSurveyReusable(false);
+  };
+
+  /**
+   * Reusable change handler
+   * 
+   * @param updatedReusable updated reusable
+   */
+  const onMaterialRowChange = async (newReusable: Reusable) => {
+    if (!keycloak?.token || !newReusable.id || !surveyId || newReusable.componentName === "") {
+      return;
+    }
+
+    try {
+      const updatedReusable = await Api.getSurveyReusablesApi(keycloak.token).updateSurveyReusable({
+        surveyId: surveyId,
+        reusableId: newReusable.id,
+        reusable: newReusable
+      });
+
+      setSurveyReusables(surveyReusables.map(reusable => (reusable.id === updatedReusable.id ? updatedReusable : reusable)));
+    } catch (error) {
+      errorContext.setError(strings.errorHandling.reusables.update, error);
+    }
+    
+    setReusableDescriptionDialogOpen(true); // TODO: Find another way to set description dialog open
+  };
+
+  /**
+   * Event handler for delete survey reusable confirm
+   */
+  const onDeleteSurveyReusableConfirm = async () => {
+    if (!keycloak?.token || !selectedReusableIds || !surveyId) {
+      return;
+    }
+
+    const reusablesApi = Api.getSurveyReusablesApi(keycloak.token);
+
+    await Promise.all(
+      selectedReusableIds.map(async materialId => {
+        await reusablesApi.deleteSurveyReusable({
+          surveyId: surveyId,
+          reusableId: materialId.toString()
+        });
+      })
+    );
+
+    fetchSurveyReusables();
+    setSelectedReusableIds([]);
+    setDeletingMaterial(false);
+  };
+
+  /**
+   * Event handler for new material change
+   * @param event event
+   */
+  const onNewMaterialChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value, name } = event.target;
+
+    switch (name) {
+      case "usability":
+        setNewMaterial({ ...newMaterial, [name]: value as Usability });
+        break;
+      case "unit":
+        setNewMaterial({ ...newMaterial, [name]: value as Unit });
+        break;
+      case "amount":
+        setNewMaterial({ ...newMaterial, [name]: Number(value) });
+        break;
+      default:
+        setNewMaterial({ ...newMaterial, [name]: value });
+        break;
+    }
+  };
```

Comment:
This is not the most readable way of doing it. I'd create two separate handler for strings and numbers. This way you don't need to as type casts to variables:
```suggestion
  /**
   * Event handler for new material string change
   *
   * @param event React change event
   */
  const onNewMaterialTextChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value, name } = event.target;

    setNewMaterial({ ...newMaterial, [name]: value });
  };

  /**
   * Event handler for new material number change
   *
   * @param event React change event
   */
  const onNewMaterialNumberChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value, name } = event.target;

    setNewMaterial({ ...newMaterial, [name]: Number(value) });
  };
```